File: c++/RotateImage.cpp
// Rotate Image LeetCode - 48

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int m= matrix.size();
        int n= m;

        // generate transpose
        for(int i=0;i<m;++i){
            for(int j=i;j<n;++j){
                swap(matrix[i][j],matrix[j][i]);
            }
        }

        //reverse rows
        for(int i=0;i<m;++i){
            reverse(matrix[i].begin(),matrix[i].end());
        }
    }
};

File: JavaScript/RotateImage.js
// Rotate Image LeetCode - 48

const rotate = function (matrix) {
  const m = matrix.length;
  const n = m;

  // make tanspose matrix
  for (let i = 0; i < m; ++i) {
    for (let j = i; j < n; ++j) {
      const tmp = matrix[i][j];
      matrix[i][j] = matrix[j][i];
      matrix[j][i] = tmp;
    }
  }

  //reverse rows
  for (let i = 0; i < m; ++i) {
    matrix[i].reverse();
  }
};
